rasterio/_env.pyx

```diff
@@ -59,12 +59,22 @@ def driver_count():
     return GDALGetDriverCount() + OGRGetDriverCount()
 
 
-cpdef get_gdal_config(key):
-    """Get the value of a GDAL configuration option"""
-    key = key.upper().encode('utf-8')
```

Comment:
@sgillies I removed the `upper()` call in all the GDAL config accessors since apparently GDAL does that on its own.  I also added a canary test that will fail if GDAL ever changes that behavior.  It could be an important change that affects how Rasterio discovers existing environment variables.  There's a solid argument that this should be its own PR, but its all in a single commit that should be easy to remove if needed.